src/exe/colmap.cc

```diff
@@ -499,12 +499,16 @@ bool VerifyCameraParams(const std::string& camera_model,
   return true;
 }
 
+enum class CameraMode { AUTO = 0, SINGLE = 1, PER_FOLDER = 2, PER_IMAGE = 3 };
+
 int RunFeatureExtractor(int argc, char** argv) {
   std::string image_list_path;
+  int camera_mode = (int)CameraMode::AUTO;
 
   OptionManager options;
   options.AddDatabaseOptions();
   options.AddImageOptions();
+  options.AddDefaultOption("camera_mode", &camera_mode);
```

Comment:
Needs documentation.

How does this interact with the following definitions in option_manager.cc?
```
AddAndRegisterDefaultOption("ImageReader.single_camera",
                              &image_reader->single_camera);
  AddAndRegisterDefaultOption("ImageReader.single_camera_per_folder",
                              &image_reader->single_camera_per_folder);
```